src/upload-request.js

```diff
@@ -0,0 +1,64 @@
+var request = require('superagent');
+var Promise = require('es6-promise').Promise;
+
+var BASE_URL = 'https://content.dropboxapi.com/2/';
+
+// This doesn't match what was spec'd in paper doc yet
+var buildCustomError = function (error, response) {
+  return {
+    status: error.status,
+    error: response.text,
+    response: response
+  };
+};
+
+var uploadRequest = function (path, args, accessToken, selectUser) {
+  var promiseFunction = function (resolve, reject) {
+    var apiRequest;
+
+    // !! This is a hack !!
+    // Since the function only takes one object for body/args right now, im
+    // doing this to make it work. I'd like to merge a branch that changes the
+    // function signature to include both body and args before landing this.
+    var body = args.body;
+    delete args.body;
```

Comment:
s/body/contents to make a more user friendly interface and hide some implementation details.